### PR TITLE
Create Incoming TeamViewer Connection Rules

### PR DESCRIPTION
**Pull request by joshnck**
### Summary of the Pull Request

TeamViewer, when set as a service, can be a little difficult to track down when a new session is created. When an incoming session is requested, the process will execute a specific command line before even writing to the incoming_connections.txt log file. 

### Changelog

new: Remote Access Tool - Team Viewer Session Started On Linux Host
new: Remote Access Tool - Team Viewer Session Started On MacOS Host
new: Remote Access Tool - Team Viewer Session Started On Windows Host

### Example Log Event

image
image

### Fixed Issues

N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Comment by joshnck** [comment]
I think this is true for Mac too - but I don't have any logs to verify.

File: rules/windows/process_creation/proc_creation_win_remote_access_tools_ultraviewer.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-title: Use of UltraViewer Remote Access Software
+title: Remote Access Tool - UltraViewer Execution
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_remote_access_tools_ultraviewer.yml:1]: We update the modified date for changes on the title

Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_remote_access_tools_ultraviewer.yml:1]: You could've update it 😛

File: rules/windows/process_creation/proc_creation_win_remote_access_tools_teamviewer_incoming_connection.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    - attack.initial_access
+    - attack.t1133
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    selection:
+        Image: 'TeamViewer_Desktop.exe'
+        ParentImage: 'TeamViewer_Service.exe'
+        CommandLine|endswith: 'TeamViewer_Desktop.exe --IPCport 5939 --Module 1'
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_remote_access_tools_teamviewer_incoming_connection.yml:26]:
```suggestion
        CommandLine|contains|all: 
            - '--IPCport 5939'
            - '--Module 1'
```
As we can switch the order

File: rules/macos/process_creation/proc_creation_macos_remote_access_tools_teamviewer_incoming_connection.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    - attack.initial_access
+    - attack.t1133
+logsource:
+    category: process_creation
+    product: macos
+detection:
+    selection:
+        ParentImage|endswith: '/TeamViewer_Service'
+        Image|endswith: '/TeamViewer_Desktop'
+        CommandLine|endswith: '/TeamViewer_Desktop --IPCport 5939 --Module 1'
```
Comment (frack113) [inline rules/macos/process_creation/proc_creation_macos_remote_access_tools_teamviewer_incoming_connection.yml:26]:
```suggestion
        CommandLine|contains|all: 
            - '--IPCport 5939'
            - '--Module 1'
```

File: rules/linux/process_creation/proc_creation_lnx_remote_access_tools_teamviewer_incoming_connection.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    - attack.initial_access
+    - attack.t1133
+logsource:
+    category: process_creation
+    product: linux
+detection:
+    selection:
+        ParentImage|endswith: '/TeamViewer_Service'
+        Image|endswith: '/TeamViewer_Desktop'
+        CommandLine|endswith: '/TeamViewer_Desktop --IPCport 5939 --Module 1'
```
Comment (frack113) [inline rules/linux/process_creation/proc_creation_lnx_remote_access_tools_teamviewer_incoming_connection.yml:26]:
```suggestion
        CommandLine|contains|all: 
            - '--IPCport 5939'
            - '--Module 1'
```

File: rules/windows/process_creation/proc_creation_win_remote_access_tools_teamviewer_incoming_connection.yml (hunk 2 above)
Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_remote_access_tools_teamviewer_incoming_connection.yml:26]: No, its an auto generated command

File: rules/windows/process_creation/proc_creation_win_remote_access_tools_ultraviewer.yml (hunk 1 above)
Comment (phantinuss) [reply rules/windows/process_creation/proc_creation_win_remote_access_tools_ultraviewer.yml:1]: not in the github UI

File: rules/windows/process_creation/proc_creation_win_remote_access_tools_teamviewer_incoming_connection.yml (hunk 2 above)
Comment (joshnck) [reply rules/windows/process_creation/proc_creation_win_remote_access_tools_teamviewer_incoming_connection.yml:26]: It is infact auto-generated by the service.

**Review by frack113 — APPROVED** [review]

**Pull request merged**